deps/src/Makefile

```diff
@@ -42,6 +42,7 @@ ifeq ($(OS), WINNT)
   SHLIB_EXT = dll 
 else ifeq ($(OS), Darwin)
   SHLIB_EXT = dylib
+  LIBS += -stdlib=libstdc++ -lc++
```

Comment:
i don't know whether this change is needed b/c of gcc or not.  just know that it works.  perhaps @btello can say.

moreover both -libstdc++ and lc++ are needed.  removing either one causes the build to fail for different reasons.  transcripts follow.

```
julia> Pkg.build("Clang")
INFO: Building Clang
g++ wrapclang.cpp.o -rdynamic -shared -o libwrapclang.dylib -L/usr/local/Cellar/llvm/3.6.2/lib  -L/usr/local/Cellar/llvm/3.6.2/lib -lclang -stdlib=libstdc++
Undefined symbols for architecture x86_64:
  "std::__1::__vector_base_common<true>::__throw_length_error() const", referenced from:
      void std::__1::vector<CXCursor, std::__1::allocator<CXCursor> >::__push_back_slow_path<CXCursor const>(CXCursor const&) in wrapclang.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [libwrapclang.dylib] Error 1
================================[ ERROR: Clang ]================================

LoadError: failed process: Process(`make`, ProcessExited(2)) [2]
while loading /Users/arthurb/.julia/v0.4/Clang/deps/build.jl, in expression starting on line 19

================================================================================

================================[ BUILD ERRORS ]================================

WARNING: Clang had build errors.

 - packages with build errors remain installed in /Users/arthurb/.julia/v0.4
 - build the package(s) and all dependencies with `Pkg.build("Clang")`
 - build a single package by running its `deps/build.jl` script

================================================================================
```

```
julia> Pkg.build("Clang")
INFO: Building Clang
g++ wrapclang.cpp.o -rdynamic -shared -o libwrapclang.dylib -L/usr/local/Cellar/llvm/3.6.2/lib  -L/usr/local/Cellar/llvm/3.6.2/lib -lclang -lc++
Undefined symbols for architecture x86_64:
  "std::terminate()", referenced from:
      ___clang_call_terminate in wrapclang.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [libwrapclang.dylib] Error 1
================================[ ERROR: Clang ]================================

LoadError: failed process: Process(`make`, ProcessExited(2)) [2]
while loading /Users/arthurb/.julia/v0.4/Clang/deps/build.jl, in expression starting on line 19

================================================================================

================================[ BUILD ERRORS ]================================

WARNING: Clang had build errors.

 - packages with build errors remain installed in /Users/arthurb/.julia/v0.4
 - build the package(s) and all dependencies with `Pkg.build("Clang")`
 - build a single package by running its `deps/build.jl` script

================================================================================
```

```
julia> Pkg.build("Clang")
INFO: Building Clang
g++ wrapclang.cpp.o -rdynamic -shared -o libwrapclang.dylib -L/usr/local/Cellar/llvm/3.6.2/lib  -L/usr/local/Cellar/llvm/3.6.2/lib -lclang -stdlib=libstdc++ -lc++

julia> 
```